### PR TITLE
chore: Update `tonic` and `prost` dependencies to latest

### PR DESCRIPTION
**Pull request by bryantbiggs**
Closes #286 
Closes #287 
Closes #288 
Closes #289

Ref: https://github.com/hyperium/hyper/issues/3110

### CODE REVIEW

File: crates/client/src/lib.rs [hunk 1]
```diff
@@ -86,23 +86,27 @@ pub async fn connect(
...
-            {
-                tokio::net::UnixStream::connect(path.clone())
-            }
-
-            #[cfg(windows)]
-            {
-                let client = tokio::net::windows::named_pipe::ClientOptions::new()
-                    .open(path.clone())
-                    .map_err(|e| std::io::Error::from(e));
-                async move { client }
```
Comment (jsturtevant) [inline crates/client/src/lib.rs:105]: does this not work with the latest versions?

Comment (bryantbiggs) [reply crates/client/src/lib.rs:105]: No - see latest commit/checks. I don't have a windows machine so I'm just guessing and checking from CI 🙈

File: crates/client/src/lib.rs [hunk 2]
```diff
@@ -86,23 +86,28 @@ pub async fn connect(
...
+                        tokio::net::UnixStream::connect(path).await?,
+                    ))
+                }
+
+                #[cfg(windows)]
+                {
+                    let client = tokio::net::windows::named_pipe::ClientOptions::new()
+                        .open(path.clone())
+                        .map_err(|e| std::io::Error::from(e));
+                    async move { client }
```
Comment (jsturtevant) [inline crates/client/src/lib.rs:109]:
```suggestion
            let client = tokio::net::windows::named_pipe::ClientOptions::new()
                .open(&path) // Provide the required argument
                .map_err(|e| std::io::Error::from(e))?;
            
            Ok::<_, std::io::Error>(hyper_util::rt::TokioIo::new(client))
```

File: crates/client/src/lib.rs (hunk 1 above)
Comment (jsturtevant) [reply crates/client/src/lib.rs:105]: got it :-), I've added a suggestion that is building on my windows machine

**Comment by ckatsak** [comment]
Hello,

I was wondering what's the status of this PR, as I came across the need for this upgrade.
Is there anything blocking this effort?

Thanks, keep up the great work :)

File: rust-toolchain.toml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.74"
```
Comment (bryantbiggs) [inline rust-toolchain.toml:2]: min version that resolved this error:

```
error: unsupported output in build script of `containerd-client v0.5.0 (/Users/runner/work/rust-extensions/rust-extensions/crates/client)`: `cargo::rerun-if-changed=vendor/github.com/containerd/containerd/api/types/descriptor.proto`
Found a `cargo::key=value` build directive which is reserved for future use.
Either change the directive to `cargo:key=value` syntax (note the single `:`) or upgrade your version of Rust.
See https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script for more information about build script outputs.
warning: build failed, waiting for other jobs to finish...
```

**Comment by bryantbiggs** [comment]
@jsturtevant any suggestions on how to deal with this https://github.com/containerd/rust-extensions/actions/runs/10822451454/job/30026310388?pr=307

**Comment by aviramha** [comment]
@bryantbiggs I think you should use `#[allow(unused_imports)]` as the intention there is probably to re-export.

**Comment by bryantbiggs** [comment]
ya, let me fire up a windows VM and try fumbling around a bit with this one so maybe we can finally get it across the line 😅

**Comment by bryantbiggs** [comment]
thanks @aviramha for the suggestion, thats what I ended up going with. Once I fired up a Windows VM there were a whole bunch of other clippy errors so just ejected out of that one 😅 

@mxpv should be ready for review now

**Review by mxpv — APPROVED** [review]
This is awesome! Thanks!

**Comment by mxpv** [comment]
Done

https://github.com/containerd/rust-extensions/releases/tag/client-v0.6.0

> On Oct 6, 2024, at 11:21 PM, Aviram Hassan ***@***.***> wrote:
> 
> @mxpv <https://github.com/mxpv> can we have a release for containerd-client please?
￼

**Pull request merged**